delivery_state/views/stock_picking_views.xml

```diff
@@ -2,20 +2,25 @@
 <!--
 Copyright 2020 Trey, Kilobytes de Soluciones
 Copyright 2020 FactorLibre
+Copyright 2020 Tecnativa - David Vidal
 License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 -->
 <odoo>
     <record id="view_picking_withcarrier_out_form" model="ir.ui.view">
         <field name="model">stock.picking</field>
         <field name="inherit_id" ref="delivery.view_picking_withcarrier_out_form"/>
         <field name="arch" type="xml">
-            <xpath expr="//group[@name='carrier_data']" position="inside">
-                <field name="date_shipped"/>
-                <field name="date_delivered"/>
-                <field name="delivery_state"/>
-                <field name="tracking_state" class="oe_inline"/>
-                <button name="tracking_state_update" string="Update tracking state" type="object" class="oe_inline"/>
-                <field name="tracking_state_history" colspan="3"/>
+            <xpath expr="//group[@name='carrier_data']" position="after">
+                <group name="delivery_tracking" attrs="{'invisible': ['|', ('state', '!=', 'done'), ('carrier_id', '=', False)]}">
+                    <field name="date_shipped"/>
+                    <field name="date_delivered"/>
+                    <field name="delivery_state"/>
+                    <field name="tracking_state" class="oe_inline"/>
+                    <button name="tracking_state_update" string="Update tracking state"
+                            type="object" class="oe_inline"
+                            attrs="{'invisible': [('state', 'in', ['customer_delivered', 'canceled_shipment'])]}" />
```

Comment:
Make it invisible also for fixed or based on rules.

Reply:
One thing the module does is to put those pickings in state _Shipping recorded in carrier_, wich maybe it's a bit misleading, as there's no possibility to push the state further... maybe for those, the state when shipped should be simply `customer_delivered`. What do you think, @rlizana ?

Reply:
I think it's OK to be on that state, as we don't know really if they are delivered or not.

Reply:
Ok then :)